src/components/indicators/z-stepper-item/index.tsx

```diff
@@ -24,33 +24,40 @@ export class ZStepperItem {
   @Prop({attribute: "pressed"})
   pressed: boolean;
 
+  /**
+   * The checked state of the item.
+   */
+  @Prop({attribute: "checked"})
+  checked: boolean;
+
   /**
    * The disabled state of the item.
    */
   @Prop({attribute: "disabled"})
   disabled: boolean;
 
-  render(): HTMLAnchorElement | HTMLButtonElement {
-    if (this.href) {
-      return (
-        <a
-          href={!this.disabled && this.href}
-          class="stepper-item"
-        >
-          <div class="indicator">{this.index}</div>
-          <span>
-            <slot />
-          </span>
-        </a>
-      );
-    }
+  private getAttributes(): Record<string, unknown> {
+    const href = this.href && !this.pressed && !this.disabled ? {onClick: () => (location.href = this.href)} : {};
+    const role = this.href && !this.disabled ? {role: "link"} : {};
+    const current = this.pressed && !this.disabled ? {ariaCurrent: "step"} : {};
+    const tabindex = this.pressed || !this.href ? {tabindex: -1} : {};
```

Comment:
Se `pressed=true` hai già `href={}`, quindi in teoria dovrebbe bastare 
`const tabindex = !this.href ? {tabindex: -1} : {};`
no?

Reply:
In realtà no, nel senso che:`!this.href` ma anche `!href`  danno sempre `false` anche quando l'oggetto è vuoto. Bisognerebbe calcolarlo  con  `Object.keys(href).length` .
Ma al netto del controllo ho preferito rendere ogni valore indipendente, in modo che se dovesse cambiare il modo di calcolare l'href, non cambia la gestione del tabindex.